lib/date_processor.rb

```diff
@@ -7,6 +7,7 @@ def initialize(input_date)
     @offset_shift_keys = "abcd"
     @counter = 0
     @offset_shift_values = {}
+    process_date
```

Comment:
this was changed to have immediate access to offset values upon instantiation. Maybe make a class method?